cpp/src/arima/batched_arima.hpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <cuml/cuml.hpp>
+#include <vector>
+
+namespace ML {
+
+/* Compute the loglikelihood of the given parameter on the given time series in a batched context.
+ * @param handle cuML handle
+ * @param y series to fit: shape = (nobs, num_bathces) and expects column major data layout. Memory on Device.
+ * @param num_batches number of time series
+ * @param order ARIMA order (p: number of ar-parameters, d: difference parameter, q: number of ma-parameters)
+ * @param params parameters to evaluate group by series: [mu0, ar.., ma.., mu1, ..] Memory on device.
+ * @param d_vs The residual between model and original signal. shape = (nobs, num_batches) Memory on device.
+ * @param trans run `jones_transform` on params.
+ * @return vector of log likelihood, one for each series (size: num_batches). Memory on host.
+ * @return kalman residual, shape = (nobs, num_batches) Memory on device.
+ */
```

Comment:
A few comments regarding the doxygen (which is important to generate the documentation properly):
- Doxygen requires the block comment to start with `/**`.
- Also a `void` method shouldn't have `@return`. You can make use of `@param[in]` and `@param[out]` to express inputs and outputs in the parameters.
- The `@param` statements should match the method's parameters
- For readability it's better to limit the line length, e.g a standard is 80 characters. And I find it better to align the documentation of the parameters as a table.

The fixed doxygen for this method would be:
```c++
/**
 * Compute the loglikelihood of the given parameter on the given time series
 * in a batched context.
 * 
 * @param[in]   handle       cuML handle
 * @param[in]   d_y          Series to fit: shape = (nobs, num_batches) and
 *                           expects column major layout. Memory on Device.
 * @param[in]   num_batches  Number of time series
 * @param[in]   nobs         Number of time series
 * @param[in]   p            Number of ar-parameters
 * @param[in]   d            Difference parameter
 * @param[in]   q            Number of ma-parameters)
 * @param[in]   d_params     Parameters to evaluate group by series:
 *                           [mu0, ar.., ma.., mu1, ..] Memory on device.
 * @param[out]  loglike      Vector of log likelihood, one for each series
 *                           (size: num_batches). Memory on host.
 * @param[in]    d_vs        The residual between model and original signal.
 *                           shape = (nobs, num_batches) Memory on device.
 * @param[in]   trans        Run `jones_transform` on params.
 */
```
(this doc might be wrong since you marked a return kalman residual and I couldn't see where it is returned? In `d_vs`?)

This comment also applies for the other functions.

Additionally, maybe the input should be marked as `const` to assure that the operations don't write to it?